2_run/src/run_glm3.R

```diff
@@ -133,20 +131,41 @@ run_glm3_model <- function(sim_dir, nml_objs, model_config, burn_in, burn_out, e
           glm_time <- system.time({glm_code <- GLM3r::run_glm(sim_lake_dir, verbose = FALSE)})[['elapsed']]
           # Pull out the final date from the output to
           # check against requested simulation end date
-          output_dates <- glmtools::get_temp(nc_filepath) %>%
+          # use 1D variable 'evap' rather than 2D 'temp' to check output dates because
+          # 1) less expensive to pull and 2) defaults cause an error with glmtools::get_temp()
+          # when the lake is too shallow.
+          output_dates <- glmtools::get_var(nc_filepath, var_name = 'evap') %>%
             mutate(date = format(as.Date(lubridate::floor_date(DateTime, 'days')),"%Y-%m-%d")) %>%
             pull(date)
           max_output_date <- max(output_dates)
         },
         until=function(val, cnd) glm_code == 0 & max_output_date==sim_stop,
         max_tries = 5)
-
+      
       # make sure glm did succeed  
       if(glm_code != 0 | max_output_date!=sim_stop) stop()
       
       # extract output
       export_fl <- sprintf(export_fl_template, site_id, gcm, time_period)
-      extract_glm_output(nc_filepath, nml_obj, export_fl)
+      extraction_error <- tryCatch(
+        {
+          extract_glm_output(nc_filepath, nml_obj, export_fl)
+          extraction_error <- NA
```

Comment:
I _think_ this could just be `return(NA)`, or you should do `return(extraction_error)` under it so that it doesn't just assume its the last object created

Reply:
Didn't we test with `return()` in this first block of the `tryCatch()` and it stopped the entire function? Since the first block isn't a function like the error block?

Reply:
Oh so true. How soon I forgot!